src/Sparrow.Server/ByteString.cs

```diff
@@ -332,6 +332,17 @@ public string Substring(int length)
             return encoding.GetString(_pointer->Ptr, length);
         }
 
+        public string Substring(int offset, int length)
+        {
+            if (!HasValue)
+                return string.Empty;
+
+            EnsureIsNotBadPointer();
+
+            var encoding = Encodings.Utf8;
+            return encoding.GetString(_pointer->Ptr + offset, length);
```

Comment:
This is not a valid way to handle things. `offset` might fall inside of UTF8 mulit byte char

Reply:
Okay, I'll remove this entirely, since we want to optimize the caller function anyway (`SplitKey`)